### PR TITLE
Hash Conversion

### PR DESCRIPTION
**Pull request by crussell-figure**
Hash Conversion

* converting nhash to hash
* fixing bug where candidate/jailed validators where being filtered out
* Updated tx query to hopefully leave off unnecessary joins
* Added a couple more indices to help with performance
* Updated asset mamager list to be more accurate

closes: #44 
closes: #45 

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/provenance-io/explorer/blob/main/CONTRIBUTING.md#pr-targeting))
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Wrote unit and integration
- [x] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

**Review by iramiller — DISMISSED** [review]
Nice solution. Note that there are states other than active without managers (destroyed)... essentially the states are sequential so I have used ‘less than active’ as those with managers, greater than equal to active do not have one’. There is also a quirk in that if someone holds 100% of the coins they have control... that last one probably isn’t important to the explorer though.

File: service/src/main/kotlin/io/provenance/explorer/domain/extensions/CoinExtensions.kt [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+const val NHASH = "nhash"
+const val HASH = "hash"
+
+fun BigInteger.toHash(denom: String) =
+    (if (denom == NHASH)
+        Pair(this.toBigDecimal().divide(BigDecimal.valueOf(1000000000), 20, RoundingMode.HALF_UP), HASH)
+    else
+        Pair(this.toBigDecimal(), denom)
+    ).let { Pair(it.first.stripTrailingZeros().toPlainString(), it.second) }
```
Comment (arnabmitra) [inline service/src/main/kotlin/io/provenance/explorer/domain/extensions/CoinExtensions.kt:16]: not sure, but removing all trailing zero's kinda munges the number of decimal places hash suppports(9 in this case)

```
fun main(){
    val b = BigDecimal("799999800199992800")
    println(b.divide(BigDecimal.valueOf(1000000000), 20, RoundingMode.HALF_UP).stripTrailingZeros())
```
gives: 799999800.1999928

`ROUND_UNNECESSARY` ..half up makes me uneasy and in principle it should be ROUND_UNNECESSARY
also i would use  UNNECESSARY(BigDecimal.ROUND_UNNECESSARY);

File: service/src/main/kotlin/io/provenance/explorer/domain/extensions/CoinExtensions.kt [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+fun String.toHash(denom: String) =
+    (if (denom == NHASH)
+        Pair(this.toBigDecimal().divide(BigDecimal.valueOf(1000000000), 20, RoundingMode.HALF_UP), HASH)
+    else
```
Comment (arnabmitra) [inline service/src/main/kotlin/io/provenance/explorer/domain/extensions/CoinExtensions.kt:21]: i would recommend some simple unit tests for ^^ methods

Comment (crussell-figure) [reply service/src/main/kotlin/io/provenance/explorer/domain/extensions/CoinExtensions.kt:21]: I agree there should be some tests, and thats on my list.

File: service/src/main/kotlin/io/provenance/explorer/domain/extensions/CoinExtensions.kt (hunk 1 above)
Comment (crussell-figure) [reply service/src/main/kotlin/io/provenance/explorer/domain/extensions/CoinExtensions.kt:16]: If the hash was 0.100000000, would you rather see that or 0.1? Cuz Id rather see 0.1

**Review by channa-figure — CHANGES_REQUESTED** [review]
Everything looks good to me, but I like @arnabmitra 's suggestion of unit tests around the BigInteger.toHash extension.    This is some good work!

File: service/src/main/kotlin/io/provenance/explorer/domain/extensions/CoinExtensions.kt (hunk 1 above)
Comment (crussell-figure) [reply service/src/main/kotlin/io/provenance/explorer/domain/extensions/CoinExtensions.kt:16]: Everything else I agree with.

Comment (arnabmitra) [reply service/src/main/kotlin/io/provenance/explorer/domain/extensions/CoinExtensions.kt:16]: would you rather see 0.1 for one tx, and then 0.011 for another and then 0.00000001 for another..personally i would like to the number of decimal places supported, but idk.. i guess keep it as is 🤷

**Review by arnabmitra — APPROVED** [review]

**Pull request merged**